chore: add DRPC tailnet & cli implementation for network telemetry

ethanndickson opened the pull request:
- Adds a `TelemetrySink` interface to `tailnet.Conn` to support sending network telemetry events regardless of source.
- Adds a `TelemetrySink` implemention for `tailnetApiConnector`, to be used by the CLI.
- Adds a way to disable network telemetry from the CLI

## TODO:
- [ ] Tests

deansheather committed:
chore: add DRPC server implementation for network telemetry

johnstcn committed:
ci: test with multiple postgres versions (#13665)

- Tests now run on postgres 16 by default when run locally (can be specified with POSTGRES_VERSION)
- Adds test-go-pg-16 to test against postgres version 16
- Updates Dogfood dockerfile / nix flake to postgres version 16
- Updates docker-compose.yaml postgres tag to 16

ethanndickson committed:
chore: add DRPC tailnet & cli implementation

ethanndickson committed:
chore: add DRPC tailnet & cli implementation

ethanndickson commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/13677?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13677** <a href="https://app.graphite.dev/github/pr/coder/coder/13677?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#13675** <a href="https://app.graphite.dev/github/pr/coder/coder/13675?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ethanndickson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

kylecarbs commented on cli/ping.go:
```diff
@@ -58,6 +58,9 @@ func (r *RootCmd) ping() *serpent.Command {
 				_, _ = fmt.Fprintln(inv.Stderr, "Direct connections disabled.")
 				opts.BlockEndpoints = true
 			}
+			if r.noNetworkTelemetry {
+				opts.EnableTelemetry = true
+			}
```
Something seems off with the naming here.

ethanndickson replied:
I left a comment with the `EnableTelemetry` definition on this, the reasoning being that we default initialize `DialAgentOptions` in tests, where it makes the most sense to have network telemetry disabled.

kylecarbs replied:
I'm just saying the naming doesn't make a lot of sense. If there is no network telemetry, why would we enable telemetry?

ethanndickson replied:
oops

ethanndickson committed:
fixup

ethanndickson committed:
fixup

ethanndickson committed:
fixup

ethanndickson committed:
fixup

Emyrk committed:
chore: implement organization scoped audit log requests (#13663)

* chore: add organization_id filter to audit logs
* chore: implement organization scoped audit log requests

Emyrk committed:
chore: implement fetch all authorized templates api (#13678)

mtojek committed:
feat: expose workspace build ID to terraform-plugin-coder (#13680)

spikecurtis committed:
fix: fix workspacesdk to return error on API mismatch (#13683)

spikecurtis committed:
fix: modify workspacesdk to ask for tailnet API 2.0 (#13684)

#13617 bumped the Agent/Tailnet API minor version because it adds telemetry features.  However, we don't actually use the protocol features yet, so it's a bit obnoxious for our CLI client to ask for the newest API version.

This is particularly true of the CLI client, since that's distributed separately, so if an end user installs the latest CLI client and their organization hasn't fully upgraded, then it will fail to connect.

Since we have a release coming up and the telemetry stuff won't make it, I think we should roll back to version 2.0 until we actually implement the telemetry stuff. That way the newest release (2.13) will work with Coder servers all the way back to 2.9.

deansheather committed:
Progress

ethanndickson committed:
Merge branch '06-26-chore_add_drpc_tailnet_cli_implementation' of https://github.com/coder/coder into 06-26-chore_add_drpc_tailnet_cli_implementation
